packages/xo-web/src/xo-app/home/index.js

```diff
@@ -439,26 +446,43 @@ const NoObjects = props =>
   props.isAdmin ? <NoObjectsWithServers {...props} /> : <NoObjectsWithoutServers {...props} />
 
 @connectStore(() => {
-  const type = (_, props) => props.location.query.t || DEFAULT_TYPE
+  const getType = (_, props) => props.location.query.t || DEFAULT_TYPE
+  const getContainers = createSelector(
+    createGetObjectsOfType('host'),
+    createGetObjectsOfType('pool'),
+    (hosts, pools) => ({ ...hosts, ...pools })
+  )
+  const getItems = createSelector(getContainers, createGetObjectsOfType(getType), (containers, items) =>
+    mapValues(items, item => ({
+      ...item,
+      container: containers[item.$container || item.$pool],
+    }))
+  )
+  // work around to avoid depending on VDIs and VBDs if type isn't "VM"
```

Comment:
If you read the code for the first time, I don't think you'd understand what makes the VMs different since it doesn't explain why you need to depend on VDIs and VBDs in the first place. So I'd suggest something like "VMs are handled separately because we need to inject their `vdisUsage`", which explains the original reason. Then you can add more details about why it can't be done in the same selector, but that's secondary IMO.